### PR TITLE
added a spinner and disabled password view for when OK is pressed. En…

### PR DESCRIPTION
**Pull request by anbitasiregar**
…ables everything and hides spinner if password is incorrect.

### CODE REVIEW

File: ThreadGroup/Controllers/TGRouterAuthViewController.m [hunk 1]
```diff
@@ -38,6 +40,9 @@ - (void)resetView {
 #pragma mark - IBActions
 
 - (IBAction)okButtonPressed:(UIButton *)sender {
+    self.view.userInteractionEnabled = NO;
```
Comment (luquantan) [inline ThreadGroup/Controllers/TGRouterAuthViewController.m]: Disabling userInteraction for the entire view works, however, we will need to cancel any pending request by pressing the "cancel" button(not implemented yet). Disable just the "ok" button instead.

File: ThreadGroup/Controllers/TGRouterAuthViewController.m [hunk 2]
```diff
@@ -28,6 +29,7 @@ - (void)viewWillAppear:(BOOL)animated {
 
 - (void)resetView {
     self.errorLabel.hidden = YES;
```
Comment (luquantan) [inline ThreadGroup/Controllers/TGRouterAuthViewController.m:32]: You need to re-enable userInteraction here. The user can return to this view and chose another router. The buttons will not function after the first successful joining request.

Comment (luquantan) [inline ThreadGroup/Controllers/TGRouterAuthViewController.m:32]: An additional thing you could do is highlight the button when it is disabled. Clearly shows to the user that the button is disabled.

File: ThreadGroup/Controllers/TGRouterAuthViewController.m (hunk 1 above)
Comment (ButkiewiczP) [inline ThreadGroup/Controllers/TGRouterAuthViewController.m]: I'm going to piggy back this comment. There's a few additional things we should probably do here. 
- Disable 'OK' Button
- Disable textfield
- Dismiss keyboard

This also implies that we'll need to present the keyboard again if the request comes back as failed.

**Comment by luquantan** [comment]
Looks good. :ship:

**Pull request merged**